### PR TITLE
Removed unwanted objects from `nan` and infinites (#1799)

### PR DESCRIPTION
**Pull request by maxonfjvipon**
Closes: #1799

### CODE REVIEW

**Comment by maxonfjvipon** [comment]
@Graur please take a look

File: eo-runtime/src/test/eo/org/eolang/negative-infinity-tests.eo [hunk 1]
```diff
@@ -27,171 +27,367 @@
...
+    $.not
+      $.equal-to positive-infinity
+
+[] > negative-infinity-not-eq-nan
+  assert-that > @
+    negative-infinity
+    $.not
+      $.equal-to nan
+
+[] > negative-infinity-not-eq-int
```
Comment (Graur) [inline eo-runtime/src/test/eo/org/eolang/negative-infinity-tests.eo:54]: @maxonfjvipon Why did you decide to split one test to a small ones?

**Review by Graur — CHANGES_REQUESTED** [review]

File: eo-runtime/src/test/eo/org/eolang/negative-infinity-tests.eo (hunk 1 above)
Comment (maxonfjvipon) [reply eo-runtime/src/test/eo/org/eolang/negative-infinity-tests.eo:54]: @Graur for some reason they causes stackoverflow exception

Comment (Graur) [reply eo-runtime/src/test/eo/org/eolang/negative-infinity-tests.eo:54]: @maxonfjvipon Looks like an issue in `eo-hamcrest`. Thanks!

**Review by Graur — APPROVED** [review]
@maxonfjvipon Looks good to me
@yegor256 Please, have a look

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**